### DISCUSSION TITLE
srsRAN gNB Trace metrics does not work.

### DISCUSSION THREAD

**Discussion by Ali78k** [srsran/srsRAN_Project#424] [General Help]
Hello friends,
Hope you are doing well.

I have worked on a srsRAN Project and tried to make a docker environment for it.
In the default scenario of [srsRAN gNB with srsUE](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#zeromq-based-setup), UE took an IP on TUN interface. I expect to measure the metrics by `t` prompt on gNB shell but nothing happened. I attached a screenshot for clarification. Can you explain it for me? What is my mistake?
![srsRAN 2024-01-23 173220](https://github.com/srsran/srsRAN_Project/assets/101015446/6d56c584-077c-4b20-adc9-f3679bb0b839)

In addition, according to below command:
`sudo ip ro add 10.45.0.0/16 via 10.53.1.2
`
are there any guarantee to send data from UE to core by gNB path?

**Comment by pgawlowicz** [comment]
do you run your `docker exec` with `-it` option?
in case of docker-compose file please add those two lines to your service:
```
    stdin_open: true
    tty: true
```

**Reply by Ali78k** [reply]
thanks dear @pgawlowicz. I've already added those lines to the service declaration, but it does not work properly. I'm working on it and will inform you.

But, what about 2nd question? 
In non-containerized approach, the routing rule that we wrote, passes packets from UE to the core network. And it doesn't force packets to pass through gNB. Am I wrong? Is it OK?
In Docker environment that I wish to make, I've declared three network bridges that shown in below picture in blue. Is this configuration enough or I have to set something else?
![gNB_srsUE_zmq_flexRIC](https://github.com/srsran/srsRAN_Project/assets/101015446/28fa95e1-8db3-485f-ba97-7fdbad3b6ef9)

Any hint will be appreciated.

**Reply by pgawlowicz** [reply]
yes, your approach seems to be correct. 
Note that there is no IP layer connection between UE and GNB, only UE and Core, so UE has to pass packets to core through its 5G network interface.

**Reply by Ali78k** [reply]
thanks a lot @pgawlowicz 
I think I need to read more about ZMQ and ZMQ's role in this scenario. I'd appreciate it if you could recommend some related references.

**Reply by pgawlowicz** [reply]
We provide a ZMQ-based RF device to be used as an emulation of a real RF device (e.g., USRP). 
The real RF devices transmit and receive signal waveform (generated by the PHY layer) over the air between GNB and UE, while the ZMQ-based RF devices transmit/receive the signal waveform samples over TCP (here ZMQ) connections. 

Please check and compare those two scenarios:

1. [Real RF devices](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#over-the-air-setup)
2. [ZMQ-based RF devices](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#zeromq-based-setup)

**Reply by Ali78k** [reply]
I really appreciate your taking the time to explain things.
